R/oxcAARCalibratedDate.R

```diff
@@ -81,13 +81,138 @@ format.oxcAARCalibratedDate <- function(x, ...){
+  cal_curve_df_new_range <- diff(range(to_plot$probability))
+
+  this_bp_distribution<-NULL
+  this_bp_distribution$y <- pretty(c(x$bp+5*x$std, x$bp-5*x$std), n=20)
+  this_bp_distribution$x <- dnorm(this_bp_distribution$y,x$bp, x$std)
+  this_bp_distribution <- as.data.frame(this_bp_distribution)
+
+  this_bp_distribution$y_rescaled <- (this_bp_distribution$y - cal_curve_df_old_min) / cal_curve_df_old_range * cal_curve_df_new_range + cal_curve_df_new_min
+
+  cal_curve_df$bp_rescaled <-
+    (cal_curve_df$bp - cal_curve_df_old_min) / cal_curve_df_old_range * cal_curve_df_new_range + cal_curve_df_new_min
+
+  cal_curve_df$sigma_rescaled <-
+    cal_curve_df$sigma / cal_curve_df_old_range * cal_curve_df_new_range
+
+  m <- ggplot() + theme_light()
+
+  graph <- m +
+    geom_area(data = to_plot, aes(x=.data$dates,
+                                  y=.data$probability,
+                                  group = .data$class,
+                                  alpha = .data$class),
+              fill = "#fc8d62",
+              position = "identity",
+              color="#00000077"
+              ) +
+    labs(title = paste0(x$name,
+                        ": ",
+                        x$bp,
+                        "\u00B1",
+                        x$std),
+         caption = x$cal_curve$name,
+         x = "Calibrated Date")  +
+    ggplot2::scale_alpha_manual(values = c(0.75, 0.25), guide = FALSE)
+
+  graph <- graph +
+    geom_ribbon(data = cal_curve_df, aes(x = .data$bc,
+                                         ymax = .data$bp_rescaled + .data$sigma_rescaled,
+                                         ymin = .data$bp_rescaled - .data$sigma_rescaled),
+                color = "#8da0cb",
+                fill = "#8da0cb",
+                alpha = 0.5) +
+    scale_y_continuous("Probability",
+                       sec.axis = sec_axis(~ (. - cal_curve_df_new_min)/ cal_curve_df_new_range * cal_curve_df_old_range + cal_curve_df_old_min,
+                                           name = "BP",
+                                           breaks=pretty(cal_curve_df$bp)),
+                       position = "right")
+
+  x_extend <- ggplot_build(graph)$layout$panel_scales_x[[1]]$range$range
+
+  this_bp_distribution$x_rescaled <- this_bp_distribution$x / max(this_bp_distribution$x) * diff(x_extend)/4 + x_extend[1]
+
+  graph <- graph +
+    geom_polygon(data = this_bp_distribution, aes(x=.data$x_rescaled, y=.data$y_rescaled),
+                 fill = "#66c2a5",
+                 alpha=0.5) +
+    scale_x_continuous(limits=x_extend, expand = c(0,0))
+
+  this_sigma_ranges <- x$sigma_ranges
+  this_sigma_qualifier <- "unmodelled"
+  if(all(!(is.null(x$posterior_sigma_ranges)), !(is.na(x$posterior_sigma_ranges)))) {
+    this_sigma_ranges <- x$posterior_sigma_ranges
+    this_sigma_qualifier <- "modelled"
+  }
+
+  sigma_text<-paste(this_sigma_qualifier,
+                    oxcAAR:::formatFullSigmaRange(this_sigma_ranges$one_sigma,"one sigma"),
```

Comment:
Just as the CRAN checks say:
```
  There are ::: calls to the package's namespace in its code. A package
    almost never needs to use ::: for its own objects
```
Usually you can call unexported package functions without a namespace definition, so only `formatFullSigmaRange()` is fine. Or was this for development? I recommend `devtools::load_all()` in that case.

Reply:
I thought, I had a good reason for this... But I removed it, works also like this, so consider it as solved.